sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/EnsureRequirements.scala

```diff
@@ -607,13 +629,14 @@ case class EnsureRequirements(
 
       case operator: SparkPlan =>
         val reordered = reorderJoinPredicates(operator)
+        val unwrapped = unwrapCastInJoinPredicates(reordered)
```

Comment:
Shall we unwrap first instead of renaming all `reorderd` variable occurrences to `unwrapped`?
```scala
- val reordered = reorderJoinPredicates(operator)
- val unwrapped = unwrapCastInJoinPredicates(reordered)
+ val unwrapped = unwrapCastInJoinPredicates(operator)
+ val reordered = reorderJoinPredicates(unwrapped)
```

Reply:
OK. The order does not affect the affected query.